siibra/features/feature.py

```diff
@@ -431,3 +471,170 @@ def __getattr__(self, __name: str):
+            [f.anchor for f in siibra_tqdm(
+                features, desc="Combining anchors", disable=logger.level < 30
+            )]
+        )
+        Feature.__init__(
+            self,
+            modality=modality,
+            description=f"Compound of {len(features)} {type(features[0])}",
+            anchor=combined_anchor,
+            datasets=list({ds for f in features for ds in f.datasets})
+        )
+        self._data_cached = None  # only to be used for the average
+        self._queryconcept = queryconcept
+        self.is_compound = True
+
+    @property
+    def subfeature_index(self) -> List[Any]:
+        """Indices, in addition to int, by which subfeatures can be accessed"""
+        return list(self._subfeatures.keys())
+
+    @property
+    def subfeature_type(self):
+        return self._subfeature_type
+
+    @property
+    def subfeatures(self):
+        return list(self._subfeatures.values())
+
+    @property
+    def data(self):
+        if self._data_cached is None:
+            try:
+                self._data_cached = sum([f.data for f in self.subfeatures]) / len(self)
+                return self._data_cached
+            except Exception:
+                raise NotImplementedError(
+                    f"Cannot get average data for CompoundFeatures of type {self._subfeature_type}"
+                )
+        else:
+            return self._data_cached
+
+    def __len__(self):
+        return len(self._subfeatures)
+
+    def __getitem__(self, filter_spec: Any):
+        if filter_spec in self._subfeatures:
+            return self._subfeatures[filter_spec]
+
+        if isinstance(filter_spec, int):
+            return list(self._subfeatures.values())[filter_spec]
+
+        raise NotFoundException(f"{filter_spec} matched no subfeatures")
+
+    @property
+    def name(self) -> str:
+        """Returns a short human-readable name of this feature."""
+        return " ".join((
+            f"{self.__class__.__name__} of {len(self)}",
+            f"{self._subfeature_type.__name__}s ({self.modality}) anchored at",
+            str(self.anchor)
+        ))
+
+    @property
+    def id(self):
+        return "::".join((
+            "cf0",
+            f"{self._subfeature_type.__name__}",
+            self._encode_concept(self._queryconcept),
+            self.datasets[0].id if self.datasets else "nodsid",
+            md5(self.name.encode("utf-8")).hexdigest()
+        ))
+
+    @classmethod
+    def compound(
+        cls,
+        features: List['Feature'],
+        queryconcept: Union[region.Region, parcellation.Parcellation, space.Space]
+    ) -> List['CompoundFeature']:
+        """
+        Compound features of the same the same type based on their anchors.
+
+        Parameters
+        ----------
+        features
+        queryconcept
+
+        Returns
+        -------
+        List[CompoundFeature]
+        """
+        return [CompoundFeature(features, queryconcept)]
```

Comment:
```suggestion
        return [cls(features, queryconcept)]
```

More customary to use cls rather than actual class name. This is so, if any class subclassing compound feature in the future, and `.compound` is called, the actual cls will be used, rather than `CompoundFeature`